### PR TITLE
[#341] New Layout for User page (connects #341)

### PR DESCRIPTION
**Pull request by razu9861**

### CODE REVIEW

File: src/app/home/home.component.ts [hunk 1]
```diff
@@ -65,8 +65,9 @@ export class HomeComponent implements OnInit, AfterViewInit {
   }
 
   backgroundRoute() {
-    const routesWithBackground = [ 'resources' ];
-    const routesWithoutBackground = [ 'resources/add', 'resources/view' ];
+    const routesWithBackground = [ 'resources', 'users' ];
+    // Leaving the exception variable in so we can easily use this while still testing backgrounds
+    const routesWithoutBackground = [];
```
Comment (paulbert) [inline src/app/home/home.component.ts]: You should not remove `resources/add` and `resources/view` from this array.  These routes still need to be included as exceptions with a white background until their views are updated.

File: src/app/users/users.component.html [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+  <mat-icon>search</mat-icon>
+  <mat-form-field class="font-size-1">
+    <input matInput (keyup)="applyResFilter($event.target.value)" i18n-placeholder placeholder="Type text to search...">
+  </mat-form-field>
+</mat-toolbar>
+
+<div class="space-container" *ngIf="displayTable">
+  <mat-toolbar class="primary-color font-size-1">
+    <span class="toolbar-fill">
+      <h1 i18n>All Users</h1>
```
Comment (paulbert) [inline src/app/users/users.component.html]: Redundant text, I think we should remove this.

File: src/app/users/users.component.html [hunk 3]
```diff
@@ -1,21 +1,56 @@
...
+    </span>
+    <div class="margin-lr-3">
+      <form (ngSubmit)="roleSubmit(allUsers.data,selectedRole)" #rolesForm="ngForm">
+        <mat-form-field>
+          <mat-select [(ngModel)]="selectedRole" name="role">
+            <mat-option *ngFor="let role of roleList" [value]="role">{{role}}</mat-option>
+          </mat-select>
+        </mat-form-field>
+        <button mat-button type="submit"><mat-icon>library_add</mat-icon><span i18n>Add Roles</span></button>
```
Comment (paulbert) [inline src/app/users/users.component.html]: Need a margin class added to the icon to space out the text.

File: src/app/users/users.component.html [hunk 4]
```diff
@@ -31,13 +66,13 @@ <h1 i18n>Users</h1>
         <div *ngIf="!element.admin">
           <mat-icon  *ngIf="element.roles.length === 0" (click)="addRole(element)">lock</mat-icon>
```
Comment (paulbert) [inline src/app/users/users.component.html]: Let's make this a `mat-icon-button`.  That should center the icon vertically.

File: src/app/users/users.component.html [hunk 5]
```diff
@@ -31,13 +66,13 @@ <h1 i18n>Users</h1>
         <div *ngIf="!element.admin">
           <mat-icon  *ngIf="element.roles.length === 0" (click)="addRole(element)">lock</mat-icon>
           <mat-icon *ngIf="element.roles.length > 0" (click)="removeRole(element)">lock_open</mat-icon>
-          <a routerLink="/users/profile/{{element.name}}" i18n mat-raised-button color="primary">View Profile</a>
+          <a routerLink="/users/profile/{{element.name}}" mat-button i18n>View Profile
```
Comment (paulbert) [inline src/app/users/users.component.html]: Lets make this a `mat-raised-button`.  It'll make it clearer that the folder icon & the text are the same.

File: src/app/home/home.component.ts (hunk 1 above)
Comment (paulbert) [reply src/app/home/home.component.ts]: Also `users/profile` and `users/update` should be added to the `routesWithoutBackground`

File: src/app/users/users.component.html [hunk 6]
```diff
@@ -1,21 +1,56 @@
...
+      <mat-cell *matCellDef="let row">
+        <mat-checkbox (click)="$event.stopPropagation()"
+          (change)="$event ? selection.toggle(row) : null"
+          [checked]="selection.isSelected(row)">
+        </mat-checkbox>
+      </mat-cell>
+    </ng-container>
+    <ng-container matColumnDef="profile">
+      <mat-header-cell *matHeaderCellDef i18n>Profile Image</mat-header-cell>
+      <mat-cell *matCellDef="let element" (click)="select(element)">{{element._attachments}}</mat-cell>
```
Comment (paulbert) [inline src/app/users/users.component.html]: This should be an img with the profile picture as src.  We should consider renaming the profile image to always have the same name which will make setting the src easier.

File: src/app/home/home.component.ts [hunk 7]
```diff
@@ -65,14 +65,14 @@ export class HomeComponent implements OnInit, AfterViewInit {
...
-    const routesWithoutBackground = [ 'resources/add', 'resources/view' ];
+    const routesWithBackground = [ 'resources', 'users' ];
+    const routesWithoutBackground = [ 'resources/add', 'resources/view', 'users/profile', 'users/update' ];
     const isException = routesWithoutBackground
       .findIndex((route) => this.router.url.indexOf(route) > -1) > -1;
     const isRoute = routesWithBackground
       .findIndex((route) => this.router.url.indexOf(route) > -1) > -1;
     return isRoute && !isException;
-  }
+}
```
Comment (lmmrssa) [inline src/app/home/home.component.ts]: why is this indentation changed

**Comment by paulbert** [comment]
Just added a user role to an admin, which we should not allow, and a role was added to a non-selected user with the add roles button.

File: src/app/users/users.component.ts [hunk 8]
```diff
@@ -134,4 +179,8 @@ export class UsersComponent implements OnInit {
     });
   }
 
+  back() {
+    this.location.back();
```
Comment (lmmrssa) [inline src/app/users/users.component.ts]: it should route back to dashboard

**Pull request merged**